ApeWorX/ape
# fix: consolidate 'ape plugins' commands and fix issues with upgrading

### What I did

fixes: #269 
fixes: #383 

was experiencing lots of bugs and inconsistencies in `ape plugins` set of commands. This PR addresses a bunch of that.

### How I did it

Lots of refactoring.
Handle results in new class.
Use `ApePlugins` class to abstract all the forms of the plugin name.
**Combine 'add' with 'uninstall' and 'remove' with 'uninstall'**

### How to verify it

`ape plugins install solidity alchemy hardhat==0.1.0b1`
`ape plugins install hardhat --upgrade`
`ape plugins uninstall hardhat`
`ape plugins uninstall`
`ape plugins install`

Make sure `ape plugins install` installs the version in the config file, even if it is already installed.
Make sure `ape plugins uninstall` and `ape plugins install --upgrade` work.

^ All negative testing you can think of for this as well. e.g. adding plugins that are already added, trying to update to version that does not exist, etc.

### Checklist
